app/handler/DonutHttpErrorHandler.java

```diff
@@ -0,0 +1,69 @@
+package handler;
+
+import play.Configuration;
+import play.Environment;
+import play.Logger;
+import play.api.OptionalSourceMapper;
+import play.api.UsefulException;
+import play.api.routing.Router;
+import play.http.DefaultHttpErrorHandler;
+import play.libs.F;
+import play.mvc.Http;
+import play.mvc.Result;
+import play.mvc.Results;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class DonutHttpErrorHandler extends DefaultHttpErrorHandler {
+
+    private static final Logger.ALogger LOG = Logger.of(DonutHttpErrorHandler.class);
+
+    @Inject
+    public DonutHttpErrorHandler(final Configuration configuration, final Environment environment,
+                                 final OptionalSourceMapper optionalSourceMapper, final Provider<Router> provider) {
+        super(configuration, environment, optionalSourceMapper, provider);
+    }
+
+    @Override
+    protected F.Promise<Result> onBadRequest(final Http.RequestHeader requestHeader, final String message) {
+        LOG.debug("Handle onBadRequest(), {}, {}", requestHeader, message);
+        return F.Promise.<Result>pure(Results.badRequest("Bad request."));
+    }
+
+    @Override
+    public F.Promise<Result> onClientError(final Http.RequestHeader requestHeader, final int i, final String message) {
+        LOG.debug("Handle onClientError(), {}, {}", requestHeader, message);
+        return super.onClientError(requestHeader, i, message);
+    }
+
+    @Override
+    protected F.Promise<Result> onDevServerError(final Http.RequestHeader requestHeader, final UsefulException e) {
+        LOG.debug("Handle onDevServerError(), {}, {}", requestHeader, e);
+        return F.Promise.<Result>pure(Results.internalServerError("A [DEV]-Server error occurred: " + e.getMessage()));
```

Comment:
I would not expose in *any* result the exception message. It is not sth. prepared (translated) and also it has security considerations.
But if it is a "UsefulException", you can provide the ID, so we can have a look into the logs: https://www.playframework.com/documentation/2.0/api/scala/play/api/PlayException$$UsefulException.html